File: app/src/main/java/com/cn29/aac/ui/shopping/ShoppingActivityModule.java
package com.cn29.aac.ui.shopping;

import com.cn29.aac.R;
import com.cn29.aac.databinding.ActivityShoppingBinding;
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel;
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModelFactory;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Charles Ng on 20/10/2017.
 */

@Module
public class ShoppingActivityModule {

  @Provides
  ActivityShoppingBinding provideBinding(ShoppingActivity shoppingActivity) {
    return DataBindingUtil.setContentView(shoppingActivity, R.layout.activity_shopping);
  }

  @Provides
  ShoppingActivityViewModel provideVM(ShoppingActivityViewModelFactory factory,
      ShoppingActivity shoppingActivity) {
    return ViewModelProviders.of(shoppingActivity, factory)
        .get(ShoppingActivityViewModel.class);
  }
}

File: app/src/main/java/com/cn29/aac/ui/shopping/ShoppingActivityModule.kt
package com.cn29.aac.ui.shopping

import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.cn29.aac.R
import com.cn29.aac.databinding.ActivityShoppingBinding
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModelFactory
import dagger.Module
import dagger.Provides


@Module
class ShoppingActivityModule {
    @Provides
    fun provideBinding(shoppingActivity: ShoppingActivity): ActivityShoppingBinding {
        return DataBindingUtil.setContentView(shoppingActivity,
                                              R.layout.activity_shopping)
    }

    @Provides
    fun provideVM(factory: ShoppingActivityViewModelFactory,
                  shoppingActivity: ShoppingActivity): ShoppingActivityViewModel {
        return ViewModelProvider(shoppingActivity, factory)
                .get(ShoppingActivityViewModel::class.java)
    }
}